src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -2259,6 +2262,46 @@ public void setACL(final String path, List<ACL> acl, int version,
     }
 
     /**
+     *
+     * @param path
+     * @param watcher explicit watcher
+     * @param maxReturned   The maximum number of children to return
+     * @param minZkid The result will be filtered out to nodes having a zkid > minZkid
+     * @return an ordered list of child nodes, ordered by mZkid
+     * @throws KeeperException
+     * @throws InterruptedException
+     */
+    public List<PathWithStat> getChildrenPaginated(final String path, Watcher watcher, final int maxReturned, final long minZkid)
```

Comment:
1. I feel its not required to name the API with Paginated, because the argument limit and offset explains the behavior.
2. Now user needs to explicitly findout the last zxid and send it again. Instead am thinking of abstracting the next batch key through an iterator kind of API. Whats your opinion?. Below is the draft implementation details.

class BatchedIterator <E> {
  /**
   * Returns <tt>true</tt> if the iteration has more elements.
   * @return <tt>true</tt> if the iterator has more elements.
   */
  boolean hasNext() throws InterruptedException, KeeperException;

  /**
   * Returns the next element in the iteration.
   * @return the next element in the iteration.
   * @throws NoSuchElementException iteration has no more elements.
   */
  E next() throws InterruptedException, KeeperException;
}

BatchedIterator <PathWithStat> listChildren(final String path, Watcher watcher, final int batchSize, final long minZxid) throws InterruptedException, KeeperException